jlm/llvm/opt/alias-analyses/Andersen.cpp

```diff
@@ -34,113 +34,104 @@ class Andersen::Statistics final : public jlm::util::Statistics
 
   explicit Statistics(jlm::util::filepath sourceFile)
       : jlm::util::Statistics(Statistics::Id::AndersenAnalysis),
-        SourceFile_(std::move(sourceFile)),
-        NumRvsdgNodes_(0),
-        NumPointerObjects_(0),
-        NumRegistersMappedToPointerObject_(0),
-        NumSupersetConstraints_(0),
-        NumAllPointeesPointToSupersetConstraints_(0),
-        NumSupersetOfAllPointeesConstraints_(0),
-        NumHandleEscapingFunctionConstraints_(0),
-        NumFunctionCallConstraints_(0),
-        NumConstraintSolvingIterations_(0),
-        NumPointsToGraphNodes_(0),
-        NumRegisterNodes_(0),
-        NumAllocaNodes_(0),
-        NumDeltaNodes_(0),
-        NumImportNodes_(0),
-        NumLambdaNodes_(0),
-        NumMallocNodes_(0),
-        NumMemoryNodes_(0),
-        NumEscapedNodes_(0),
-        NumExternalMemorySources_(0)
+        SourceFile_(std::move(sourceFile))
   {}
 
   void
   StartAndersenStatistics(const jlm::rvsdg::graph & graph) noexcept
   {
-    NumRvsdgNodes_ = jlm::rvsdg::nnodes(graph.root());
-    AnalysisTimer_.start();
-    SetAndConstraintBuildingTimer_.start();
+    AddUInteger("#RvsdgNodes", jlm::rvsdg::nnodes(graph.root()));
+    StartTimer("AnalysisTimer");
+    StartTimer("SetAndConstraintBuildingTimer");
```

Comment:
Could factor out these magic strings into variables? In particular, those strings that are used multiple times, such as "SetAndConstraintBuildingTimer"?

Reply:
I'm a bit hesitant to put all the strings into variables. Some of the purpose of this PR was to reduce the number of separate lines of code that need to be modified whenever a statistic is added, removed or changed.

If we were to add variables for these strings, they would either only be accessible within the `.cpp` file, where most of them only see one use, or they would have to be in the `.hpp` file, which makes it somewhat easier for tests to retrieve measurements, but now requires you to modify two files whenever a statistic is added/removed/modified.

The only place I can see where having access to the strings is useful is when retrieving statistics in tests. By using variables instead of magic strings, IDEs can correctly perform renaming across files, which is nice. However, a test failure makes itself apparent "almost" as quickly as a compilation error, so I don't think it is worth it to place all string literals in variables in header files for only tests' sake.

For the timers, I agree, it is a bit annoying having to reference them by name twice. It will assert false if there is a mismatch, but variables would make it clearer. The question is if the variable should be a reference to the timer object, or a string.

Reply:
I should have been a little bit more elaborate in my original comment. Sorry! Why I mentioned the magical strings was that you are requesting timers multiple times (at least twice, once for start and once for stop), and in both cases you are using a string. Thus, instead of writing something like this:

```
class Statistics
{
public:
  void
  StartMySpecialStatistics()
  {
    StartTimer("MySpecialStatisticsTimer");
    ...
  }

  void
  StopMySpecialStatistics()
  {
     StopTimer("MySpecialStatisticsTimr"); // typo here
  }
}
```

I have something like this in mind:
```
class Statistics
{
private:
  const char * MySpecialStatisticsTimerTag_ = "MySpecialStatisticsTimer";
public:
  void
  StartMySpecialStatistics()
  {
    StartTimer(MySpecialStatisticsTimerTag_);
    ...
  }

  void
  StopMySpecialStatistics()
  {
     StopTimer(MySpecialStatisticsTimerTag_);
  }
}
```

This was only meant for:
1. The statistics class internally and was not supposed to "leak" into tests/outside.
2. Only for instances were you have to use them multiple times (to avoid typos, and make your life easier in terms of renaming by being able to utilize the IDE)
3. The idea was to avoid typos and potentially reference a wrong timer, but looking at the code you would throw an exception if one would have a typo. So this argument does not really count any longer (my bad, should have taken a closer look at the code).

I do not really see that the "additional number of lines" argument counts here as most of the renaming with those tags would be done by the IDE in the end. Personally, I would still try to use such tags for entities that I have to reference at least multiple times, but this is just personal preference.

Due to the exception, the original argument is void. Leave it as it is if you like. No strong opinion about this.